src/main/resources/public/modules/windflowUtils.js

```diff
@@ -548,6 +548,12 @@ export function makeContextEditMode({
         area.chapters = chapterIds.map(chapterId => area.chapters.find(chapter => chapter.id === chapterId));
     };
 
+    const removeChapter = (areaName, chapterId) => {
+        let area = state.editedPage.areas[areaName];
+        let arrayWithItemRemoved = area.chapters.filter(item => {return item.id != chapterId})
+        // What now
+    }
```

Comment:
```suggestion
    const removeChapter = (areaName, chapterId) => {
        let area = state.editedPage.areas[areaName];
        area.chapters = area.chapters.filter(chapter => chapter.id !== chapterId)
     }
```

Reply:
@maoberlehner This is what I thought and tried, but I get a strange error back. Could be because it's proxied.

![image](https://user-images.githubusercontent.com/1888768/99527293-3c1fed00-29a5-11eb-82f8-81cd07d9a21f.png)

Reply:
How is this error strange? 😅 

You remove the currently edited chapter and the app complains that it can't find the chapter you just removed (that is still set as currently edited chapter) 😉 

You must set a different `state.editedChapterId` (or `null`) before deleting a chapter.

https://github.com/windflow-io/EternalEngine/blob/master/src/main/resources/public/modules/windflowUtils.js#L497

Reply:
Ha ha. Okay. I think I understand. Standby.

Reply:
@maoberlehner I need a bit more help please. Why it's being called twice? And why is it also being called on page load?

It works perfectly if I comment out the throw. Committed maybe you can look at it and accept the pull request.

```javascript
    const removeChapter = (areaName, chapterId) => {
        console.log("Removing chapter id " + chapterId)
        let area = state.editedPage.areas[areaName];
        area.chapters = area.chapters.filter(chapter => chapter.id !== chapterId)
        console.log("Setting state.editedChapterId to null")
        state.editedChapterId = null;
    }
```

```javascript
    const editedChapter = computed(() => {
        console.log("Edited chapter clicked");
        console.log ("chapterId: " + state.editedChapterId);
        if (!state.editedChapterId) return null;

        // REFACTOR
        // Abstraction for such for of loops everywhere.
        for (const { chapters } of Object.values(state.editedPage.areas)) {
            for (const chapter of chapters) {
                if (chapter.id === state.editedChapterId) return chapter;
            }
        }

        //throw new Error(`Chapter with ID "${state.editedChapterId}" not found!`);
    });
```

![image](https://user-images.githubusercontent.com/1888768/99530612-3973c680-29aa-11eb-9a59-219416c00871.png)

Reply:
`editedChapter` is a `computed` property. The callback function is triggered whenever some reactive value inside of it is updated: https://v3.vuejs.org/guide/reactivity-computed-watchers.html#computed-and-watch
`state.editedChapterId` is set, so the `computed` callback is triggered.

> It works perfectly if I comment out the throw.

🤔 

But that is an impossible state. Maybe it does not lead to an error at the moment but if `state.editedChapterId` points to a chapter ID that does not exist anymore, this is a problem...

I think if you set `state.editedChapterId = null;` **before** removing the chapter, it should work.

Reply:
There is definitely some kind of race condition going on here. I have a feeling it's something to do with a delay in the processing of a proxy object (which I don't understand yet). What I have done is made it print a console.err instead of throw it and I'll look into it when I have more time. Key priority for me now is to get mimo out.

Please merge the pull request if you think this is acceptable.

Thanks Markus!!!!